frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -248,6 +248,31 @@ function ReaderHighlight:onTapXPointerSavedHighlight(ges)
     end
 end
 
+function ReaderHighlight:updateHighlight(page, index, move_from_beginning, dir)
+    local highlight = self.view.highlight.saved[page][index]
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    if move_from_beginning then
+        if dir == 1 then
+            local updated_highlight = self.ui.document:getNextVisibleWordEnd(highlight_beginning)
+            self.view.highlight.saved[page][index].pos0 = updated_highlight
+        else
+            local updated_highlight = self.ui.document:getPrevVisibleWordEnd(highlight_beginning)
+            self.view.highlight.saved[page][index].pos0 = updated_highlight
```

Comment:
It feels like, for the start of the selection, you might better use getNextVisibleWordStart and getPrevVisibleWordStart. These get*WordEnd would move it to the end of the previous word (what if there is none? if it is the 2nd word of a book and you want to extend it to the 1st word of the book?) And I guess you would grab at least the space between words with that, if not the last char of the previous word?